packages/postgress/src/MessagePickupRepositoryPg.ts

```diff
@@ -0,0 +1,616 @@
+      const messagesToSend = await this.messagesCollection?.query(
+        'SELECT * FROM queuedmessages WHERE state = $1 and connectionid = $2',
+        ['sending', connectionID],
+      )
+      if (messagesToSend && messagesToSend.rows.length > 0) {
+        for (const message of messagesToSend.rows) {
+          // Update the message state to 'pending'
+          await this.messagesCollection?.query('UPDATE queuedmessages SET state = $1 WHERE id = $2', [
+            'pending',
+            message.id,
+          ])
+        }
+
+        this.logger?.debug(`[checkQueueMessages] ${messagesToSend.rows.length} messages updated to 'pending'.`)
+      } else {
+        this.logger?.debug('[checkQueueMessages] No messages in "sending" state.')
+      }
+    } catch (error) {
+      this.logger?.error(`[checkQueueMessages] Error processing messages: ${error}`)
+    }
+  }
+
+  /**
+   * Get current active live mode message pickup session for a given connection
+   * @param connectionId
+   * @returns
+   */
+  private async getLocalliveSession(connectionId: string): Promise<MessagePickupSession | undefined> {
+    this.logger?.debug(`[getLocalliveSession] Verify current active live mode for connectionId ${connectionId}`)
+
+    try {
+      if (!this.agent) throw new Error('Agent is not defined')
+      return this.agent.messagePickup.getLiveModeSession({ connectionId })
+    } catch (error) {
+      this.logger?.error(`[getLocalliveSession] error in getLocalliveSession: ${error}`)
+    }
+  }
+
+  /**
+   * This method allow find record into DB to determine if the connectionID has a liveSession in another instance
+   * @param connectionId
+   * @returns liveSession object or false
+   */
+  private async getLiveSessionFromDB(connectionId: string): Promise<any | boolean> {
+    this.logger?.debug(`[getLiveSessionFromDB] initializing find registry for connectionId ${connectionId}`)
+    if (!connectionId) throw new Error('connectionId is not defined')
+    try {
+      const queryLiveSession = await this.messagesCollection?.query(
+        `SELECT * FROM storelivesession WHERE connectionid = $1 LIMIT $2`,
+        [connectionId, 1],
+      )
+      // Check if liveSession is not empty (record found)
+      const recordFound = queryLiveSession && queryLiveSession.rows && queryLiveSession.rows.length > 0
+      this.logger?.debug(`[getLiveSessionFromDB] record found status ${recordFound} to connectionId ${connectionId}`)
+      return recordFound ? queryLiveSession.rows[0] : false
+    } catch (error) {
+      this.logger?.debug(`[getLiveSessionFromDB] Error find to connectionId ${connectionId}`)
+      return false // Return false in case of an error
+    }
+  }
+
+  /**
+   * This method adds a new connectionId and instance name to DB upon LiveSessionSave event
+   * @param connectionId
+   * @param instance
+   */
+  private async addLiveSessionFromDb(session: MessagePickupSession, instance: string): Promise<void> {
```

Comment:
```suggestion
  private async addLiveSessionOnDb(session: MessagePickupSession, instance: string): Promise<void> {
```

Here is where I don't remember the need of using an instance identifier. But if it serves for debug purposes we can of course keep it (maybe as an optional parameter, and put an uuid if not defined in the config).

Reply:
The instance variable will help us determine in the database which instance it is registered to. For this reason, it was added to the schema.